cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQueryBuilder.java

```diff
@@ -0,0 +1,613 @@
+        cristinQuery.isNvaQuery = cristinQuery.isNvaQuery
+            || tmpSet.contains(PATH_ORGANISATION)
+            || tmpSet.contains(PATH_PROJECT);
+        cristinQuery.otherRequiredKeys.addAll(tmpSet);
+        return this;
+    }
+
+    /**
+     * Setter Organization.
+     */
+    public CristinQueryBuilder asNvaQuery() {
+        cristinQuery.isNvaQuery = true;
+        return this;
+    }
+
+    /**
+     * Setter code for the authority that evaluated a project approval.
+     */
+    public CristinQueryBuilder withApprovedBy(String approvedby) {
+        cristinQuery.setValue(PROJECT_APPROVED_BY, approvedby);
+        return this;
+    }
+
+    /**
+     * Setter reference id of project approval.
+     */
+    public CristinQueryBuilder withApprovalReferenceId(String approvalReferenceId) {
+        cristinQuery.setValue(PROJECT_APPROVAL_REFERENCE_ID, approvalReferenceId);
+        return this;
+    }
+
+    /**
+     * Setter Biobank id.
+     */
+    public CristinQueryBuilder withBiobank(String biobank) {
+        cristinQuery.setValue(BIOBANK, biobank);
+        return this;
+    }
+
+    /**
+     * Setter funding source code e.g: NFR, and project_code together separated by ':'
+     */
+    public CristinQueryBuilder withFunding(String funding) {
+        cristinQuery.setValue(FUNDING, funding);
+        return this;
+    }
+
+    /**
+     * Setter funding source code.
+     */
+    public CristinQueryBuilder withFundingSource(String fundingSource) {
+        cristinQuery.setValue(FUNDING_SOURCE, fundingSource);
+        return this;
+    }
+
+
+    /**
+     * Builder GRANT ID..
+     */
+    public CristinQueryBuilder withGrantId(String grantId) {
+        cristinQuery.setValue(GRANT_ID, grantId);
+        return this;
+    }
+
+    public CristinQueryBuilder withIdentity(String identifier) {
+        return withPathIdentity(identifier);
+    }
+
+    /**
+     * Setter coordinating institution by cristin id, acronym, name, or part of the name.
+     */
+    public CristinQueryBuilder withInstitution(String institution) {
+        cristinQuery.setValue(INSTITUTION, institution);
+        return this;
+    }
+
+    /**
+     * Start of pagination.
+     */
+    public CristinQueryBuilder withItemsFromPage(String page) {
+        cristinQuery.setValue(PAGE_CURRENT, page);
+        return this;
+    }
+
+    /**
+     * Items per page.
+     */
+    public CristinQueryBuilder withItemsPerPage(String itemsPerPage) {
+        cristinQuery.setValue(PAGE_ITEMS_PER_PAGE, itemsPerPage);
+        return this;
+    }
+
+    /**
+     * Setter sorting on 'start_date' and/or 'end_date'.
+     */
+    public CristinQueryBuilder withItemSort(String sort) {
+        cristinQuery.setValue(PAGE_SORT, sort);
+        return this;
+    }
+
+    /**
+     * Setter search by keyword.
+     */
+    public CristinQueryBuilder withKeyword(String keyword) {
+        cristinQuery.setValue(PROJECT_KEYWORD, keyword);
```

Comment:
Are these values in this class set regardless of them being supplied or not?

Reply:
only if they are defined in parameters and have a valid key(name)

```
    public CristinQueryBuilder fromQueryParameters(Map<String, String> parameters) {
        parameters.forEach(this::setValue);
        return this;
    }
```